### PR TITLE
searchCompany

### PR DESCRIPTION
**Pull request by LinhHoang1110**
## Related Tickets:
- [#28758](https://edu-redmine.sun-asterisk.vn/issues/28758)

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.
searchCompany1

### CODE REVIEW

**Comment by LinhHoang1110** [comment]
ready

File: resources/views/layouts/app.blade.php [hunk 1]
```diff
@@ -20,6 +20,7 @@
     <link rel="stylesheet" type="text/css" href="/bower_components/demo-bower/assets/lib/slick/slick-theme.css">
     <link rel="stylesheet" type="text/css" href="/bower_components/demo-bower/assets/css/style.css">
     <link rel="stylesheet" type="text/css" href="/bower_components/demo-bower/assets/css/responsive.css">
+    <link rel="stylesheet" href="{{ asset('/css/home.css') }}">
     <title>Document</title>
```
Comment (thaonp104) [inline resources/views/layouts/app.blade.php]: i18n

File: resources/views/layouts/app.blade.php [hunk 2]
```diff
@@ -32,10 +33,11 @@
                             <a href="" title=""><img src="{{ config('icon.logo') }}" alt=""></a>
                         </div>
                         <div class="search-bar">
-                            <form>
-                                <input type="text" name="search" placeholder="Search...">
+                            <form action="#">
+                                <input id="inputSearchCompany" type="text" name="search" placeholder="Search...">
```
Comment (thaonp104) [inline resources/views/layouts/app.blade.php]: i18n

File: resources/views/layouts/app.blade.php [hunk 3]
```diff
@@ -217,10 +219,6 @@ class="dropdown-item" href="#">
     <script type="text/javascript" src="/bower_components/demo-bower/assets/js/flatpickr.min.js"></script>
     <script type="text/javascript" src="/bower_components/demo-bower/assets/lib/slick/slick.min.js"></script>
     <script type="text/javascript" src="/bower_components/demo-bower/assets/js/script.js"></script>
-    <script>
-        document.querySelector('#btn-logout').onclick = function () {
-            document.querySelector('#logout-form').submit();
-        }
-    </script>
+    <script src="{{ asset('/js/home.js') }}"></script>
```
Comment (thaonp104) [inline resources/views/layouts/app.blade.php]:
```suggestion
    <script src="{{ asset('js/home.js') }}"></script>
```

**Comment by LinhHoang1110** [comment]
ready

File: app/Http/Controllers/searchCompanyController.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Company;
+use Illuminate\Http\Request;
+
+class searchCompanyController extends Controller
```
Comment (huyenttm-0901) [inline app/Http/Controllers/searchCompanyController.php]:
```suggestion
class SearchCompanyController extends Controller
```

File: resources/js/home.js [hunk 5]
```diff
@@ -1,4 +1,33 @@
...
+                data.map(result => {
+                    if (inputSearch.value === '') {
+                        resultSearch.innerHTML = '';
+                    } else {
+                        resultSearch.innerHTML = `<a href="#">${result.name}</a>`;
+                    }
+                })
+            },
+            error: function (err) {
+                console.log(err)
```
Comment (huyenttm-0901) [inline resources/js/home.js]: bỏ console.log

File: resources/js/home.js [hunk 6]
```diff
@@ -1,4 +1,33 @@
...
+inputSearch.addEventListener('keyup', () => {
+    setTimeout(() => {
+        $.ajax({
+            type: 'get',
+            url: '/getCompaniesFromSearch',
+            data: {
+                data : inputSearch.value,
+            },
+            success: function (data) {
+                console.log(data);
```
Comment (huyenttm-0901) [inline resources/js/home.js]: tương tụ

File: webpack.mix.js [hunk 7]
```diff
@@ -15,3 +15,4 @@ mix.js('resources/js/app.js', 'public/js')
     .sass('resources/sass/app.scss', 'public/css');
 
 mix.js('resources/js/home.js', 'public/js');
+mix.copyDirectory('resources/sass/home.css', 'public/css');
```
Comment (huyenttm-0901) [inline webpack.mix.js]: sao em lại copy ở đây? mĩ bằng style chứ

**Review by huyenttm-0901 — COMMENTED** [review]
thêm attachment vào pull

**Comment by LinhHoang1110** [comment]
> ## Related Tickets:
> * [#28758](https://edu-redmine.sun-asterisk.vn/issues/28758)
> 
> ## WHAT this PR does?
> ## Notes Impacted Areas:
> * N/A.
> 
> ## Attachments:
> * N/A.

**Comment by LinhHoang1110** [comment]
ready

**Pull request merged**